### PR TITLE
EPMDJ-6320: Options fix address format validation

### PR DESCRIPTION
**Pull request by RomanAchilov**
Issue: EPMDJ-6320:

### CODE REVIEW

File: __tests__/unit/validate-backend-adress.spec.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+    expect(validator({ backendAddress: 'http://my.example.fr:-1234' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://my.example.fr.' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://WWW.exaMple.FR' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://what..ever.com' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://what.ever.com..' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://www.пример.bg' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://...' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://..' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://.' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://.' })).toEqual(error);
```
Comment (RomanAchilov) [inline __tests__/unit/validate-backend-adress.spec.ts:23]: delete

File: src/options-script/app.tsx [hunk 2]
```diff
@@ -75,13 +62,33 @@ export const App = optionsPage(({ className }: Props) => {
...
+            const prevValue = prevValueRef.current;
             return (
               <form className="d-flex flex-column gy-4">
                 <FormGroup label="Admin API URL">
                   <Field
                     name="backendAddress"
                     component={Fields.Input}
                     placeholder="protocol://hostname:port"
+                    parse={(value: string) => {
+                      const hostWithHttpProtocol = () => {
```
Comment (RomanAchilov) [inline src/options-script/app.tsx:79]: change name

File: __tests__/unit/validate-backend-adress.spec.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    expect(validator({ backendAddress: 'http://..' })).toEqual(error);
+    expect(validator({ backendAddress: 'http://.' })).toEqual(error);
+  });
+
+  it('should return no error if backendAddress is valid', () => {
+    expect(validator({ backendAddress: 'http://foo.ru' })).toBeUndefined();
+    expect(validator({ backendAddress: 'http://localhost' })).toBeUndefined();
+    expect(validator({ backendAddress: 'http://www.localhost.ru' })).toBeUndefined();
+    expect(validator({ backendAddress: 'http://localhost.uk' })).toBeUndefined();
+    expect(validator({ backendAddress: 'http://0.0.0.0:0' })).toBeUndefined();
```
Comment (RomanDavlyatshin) [inline __tests__/unit/validate-backend-adress.spec.ts:30]: It's not super-important, but I guess we should restrict entering '0.0.0.0:0'. It's a valid IP address, but it doesn't make much sense to allow a user to enter it.
References:
https://www.howtogeek.com/225487/what-is-the-difference-between-127.0.0.1-and-0.0.0.0/
https://en.wikipedia.org/wiki/0.0.0.0#:~:text=0.0%20is%20used%20with%20a,directed%20towards%20a%20local%20router.

File: src/forms/form-validators.ts [hunk 4]
```diff
@@ -68,11 +68,24 @@ export function toError(fieldName: string, error: string) {
...
+  }
+}
+
+export function validateBackendAdress(fieldName: string): FormValidator {
   return (valitationItem) => {
     const value = get<string>(valitationItem, fieldName) || '';
-    return value.replace(pattern, '')
-      ? toError(fieldName, errorMessage)
-      : undefined;
+    if (value === 'localhost' || value.slice(-3) === ':80' || value.slice(-8) === ':0008080') return undefined;
```
Comment (RomanDavlyatshin) [inline src/forms/form-validators.ts:87]: `http://пример:80`

**Pull request merged**